### ISSUE TITLE
Error when running cellphonedb on too many cells

### ISSUE THREAD

**Issue by wzhao01** [Teichlab/cellphonedb#136]
Hi,

I'm receiving the following unexpected error in the 'Running Statistical Analysis' step:

 File "/Linux/redhat_7_x86_64/pkgs/anaconda3_5.3.1/lib/python3.7/multiprocessing/pool.py", line 657, in get
    raise self._value
multiprocessing.pool.MaybeEncodingError: Error sending result: 
...
...
[1364 rows x 3481 columns]]'. Reason: 'error("'i' format requires -2147483648 <= number <= 2147483647")'

This error only occurs when I am running cellphonedb on larger datasets.

Thank you for your help!

**Comment by lijxug** [comment]
I've encountered the same issue. Does anyone have a solution?

**Comment by brianpenghe** [comment]
I also met this problem
```
Traceback (most recent call last):
  File "/home/ubuntu/miniconda3/envs/cpdb/lib/python3.7/site-packages/cellphonedb/src/api_endpoints/terminal_api/metho
d_terminal_api_endpoints/method_terminal_commands.py", line 144, in statistical_analysis
    subsampler,
  File "/home/ubuntu/miniconda3/envs/cpdb/lib/python3.7/site-packages/cellphonedb/src/local_launchers/local_method_lau
ncher.py", line 64, in cpdb_statistical_analysis_local_method_launcher
    subsampler
  File "/home/ubuntu/miniconda3/envs/cpdb/lib/python3.7/site-packages/cellphonedb/src/core/methods/method_launcher.py"
, line 76, in cpdb_statistical_analysis_launcher
    self.separator)
  File "/home/ubuntu/miniconda3/envs/cpdb/lib/python3.7/site-packages/cellphonedb/src/core/methods/cpdb_statistical_an
alysis_method.py", line 36, in call
    result_precision,
  File "/home/ubuntu/miniconda3/envs/cpdb/lib/python3.7/site-packages/cellphonedb/src/core/methods/cpdb_statistical_analysis_complex_method.py", line 97, in call
    separator)
  File "/home/ubuntu/miniconda3/envs/cpdb/lib/python3.7/site-packages/cellphonedb/src/core/methods/cpdb_statistical_analysis_helper.py", line 303, in shuffled_analysis
    results = pool.map(statistical_analysis_thread, range(iterations))
  File "/home/ubuntu/miniconda3/envs/cpdb/lib/python3.7/multiprocessing/pool.py", line 268, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/home/ubuntu/miniconda3/envs/cpdb/lib/python3.7/multiprocessing/pool.py", line 657, in get
    raise self._value
multiprocessing.pool.MaybeEncodingError: Error sending result: '[       

[1296 rows x 20736 columns]]'. Reason: 'error("'i' format requires -2147483648 <= number <= 2147483647")'

```

**Comment by brianpenghe** [comment]
Just got help from my labmate @nh3 👍 

> https://stackoverflow.com/questions/47776486/python-struct-error-i-format-requires-2147483648-number-2147483647

Alternatively, you could use cellphonedb v2.1.8b3 which is available on pip. See https://github.com/Teichlab/cellphonedb/pull/288#pullrequestreview-654402223

That version contains a fix that get around that issue.

**Comment by Zifeng-L** [comment]
> Just got help from my labmate @nh3 👍
> 
> > https://stackoverflow.com/questions/47776486/python-struct-error-i-format-requires-2147483648-number-2147483647
> 
> > Alternatively, you could use cellphonedb v2.1.8b3 which is available on pip. See [#288 (review)](https://github.com/Teichlab/cellphonedb/pull/288#pullrequestreview-654402223)
> 
> > That version contains a fix that get around that issue.

Hi here, I just want to know if I should modify the source code of cellphoneDB manually?

**Comment by nh3** [comment]
> > Just got help from my labmate @nh3 👍
> > > https://stackoverflow.com/questions/47776486/python-struct-error-i-format-requires-2147483648-number-2147483647
> > 
> > 
> > > Alternatively, you could use cellphonedb v2.1.8b3 which is available on pip. See [#288 (review)](https://github.com/Teichlab/cellphonedb/pull/288#pullrequestreview-654402223)
> > 
> > 
> > > That version contains a fix that get around that issue.
> 
> Hi here, I just want to know if I should modify the source code of cellphoneDB manually?

Which version of CellphoneDB was used? This issue should be fixed in v2.1.8 or above.
Also, active development (v3 and above) has moved to https://github.com/ventolab/CellphoneDB.

**Comment by Zifeng-L** [comment]
> > > Just got help from my labmate @nh3 👍
> > > > https://stackoverflow.com/questions/47776486/python-struct-error-i-format-requires-2147483648-number-2147483647
> > > 
> > > 
> > > > Alternatively, you could use cellphonedb v2.1.8b3 which is available on pip. See [#288 (review)](https://github.com/Teichlab/cellphonedb/pull/288#pullrequestreview-654402223)
> > > 
> > > 
> > > > That version contains a fix that get around that issue.
> > 
> > 
> > Hi here, I just want to know if I should modify the source code of cellphoneDB manually?
> 
> Which version of CellphoneDB was used? This issue should be fixed in v2.1.8 or above. Also, active development (v3 and above) has moved to https://github.com/ventolab/CellphoneDB.

Doing `conda install python=3.8` works for me! Thank you very much!
